### PR TITLE
ccm-initialise / ProfileCache: factor out the creation and setup of cache_root

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by stdweird** [comment]
This allows to change eg the group_readable permissions by modifying the cache_root.

File: src/main/bin/ccm-initialise [hunk 1]
```diff
@@ -7,131 +7,121 @@
...
+#
+# initialise
+#
+sub _initialize {
+
+    # local lexicals
+    my $self = shift;
+
+    # version and usage
+    $self->{'VERSION'} = "abc";
```
Comment (ned21) [inline src/main/bin/ccm-initialise]: Is this necessary? Can we get a real value in here somehow?

Comment (stdweird) [inline src/main/bin/ccm-initialise]: oops. some testing code got pushed. it contained the maven template (but give perl errors when testing untemplated :smile: ). will fix

**Comment by jrha** [comment]
LGTM

**Comment by stdweird** [comment]
@jrha @ned21 what are the chances of getting this merged? i'm debugging https://github.com/quattor/configuration-modules-core/issues/757, which points to non-closed filewriter in Fetch/ProfileCache, which is modified in the PR.
i can fix it in master, but will have to make different fix and resolve some rebase conflict problems when the PR gets merged.

if you think this can't get merged in 16.4, i'll make a new PR for current master
(and to be clear, the current PR does not have the fix yet, but addresses cacheroot permission issues)

**Comment by ned21** [comment]
LGTM (GH suggests it might have changed since @jrha reviewed it so I'll leave him to re-review and merge)

**Pull request merged**